fastval.go

```diff
@@ -140,6 +142,10 @@ func (val FastVal) GetFloat() float64 {
 	return *(*float64)(unsafe.Pointer(&val.rawData))
 }
 
+func (val FastVal) GetRegex() *regexp.Regexp {
```

Comment:
There needs to be an "AsRegex", which needs to check `val.dataType` and perform the appropriate conversion, then AsRegex should be used in most cases where you are doing that form of matching.